### PR TITLE
Rename storage errors

### PR DESCRIPTION
**Pull request by ma2bd**
## Motivation

The concept of `DbContext` is an implementation detail. `(KeyValue)Store` is a real name for backends.

## Proposal

rename `*ContextError` into `*StoreError`

## Test Plan

CI

### CODE REVIEW

**Review by MathieuDutSik — COMMENTED** [review]
The renaming is fine except for `MemoryError` because memory error can mean just about anything in computer science.

File: linera-storage/src/memory.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ impl MemoryStorage<crate::TestClock> {
         wasm_runtime: Option<WasmRuntime>,
         max_stream_queries: usize,
         clock: TestClock,
-    ) -> Result<Self, MemoryContextError> {
+    ) -> Result<Self, MemoryError> {
```
[the remark this answers is not in the document]
Comment (afck) [reply linera-storage/src/memory.rs:33]: `MemoryStoreError`? Or `MemoryStorageError`?

[the remark this answers is not in the document]
Comment (MathieuDutSik) [reply linera-storage/src/memory.rs:33]: `MemoryStoreError` is fine.

[the remark this answers is not in the document]
Comment (ma2bd) [reply linera-storage/src/memory.rs:33]: Ok for `MemoryStoreError`. A more ambitious appraoch is to rename the module `memory_db`

Please no "storage" in views :)

[the remark this answers is not in the document]
Comment (ma2bd) [reply linera-storage/src/memory.rs]: Alright so the PR now renames everything `*ContextError` into `*StoreError`

**Review by afck — APPROVED** [review]

**Review by MathieuDutSik — APPROVED** [review]

**Pull request merged**